opsdroid/connector/matrix/connector.py

```diff
@@ -104,111 +109,139 @@ def _process_rooms_dict(self, rooms):
+
+        if isinstance(response, nio.SyncError):
+            _LOGGER.error(
+                f"Error during initial sync: {response.message} (status code {response.status_code})"
+            )
+            return
+
+        self.connection.sync_token = response.next_batch
 
         if self.nick:
-            display_name = await self.connection.get_display_name(self.mxid)
+            display_name = await self.connection.get_displayname(self.mxid)
             if display_name != self.nick:
-                await self.connection.set_display_name(self.mxid, self.nick)
+                await self.connection.set_displayname(self.nick)
 
     async def disconnect(self):
         """Close the matrix session."""
-        await self.session.close()
+        await self.connection.close()
 
     async def _parse_sync_response(self, response):
-        self.connection.sync_token = response["next_batch"]
+        self.connection.sync_token = response.next_batch
 
         # Emit Invite events for every room in the invite list.
-        for roomid, room in response["rooms"]["invite"].items():
+        for roomid, roomInfo in response.rooms.invite.items():
             # Process the invite list to extract the person who invited us.
             invite_event = [
                 e
-                for e in room["invite_state"]["events"]
-                if "invite" == e.get("content", {}).get("membership")
+                for e in roomInfo.invite_state
+                if isinstance(e, nio.InviteMemberEvent)
+                if e.membership == "invite"
             ][0]
-            sender = await self.get_nick(None, invite_event["sender"])
+            sender = await self.get_nick(None, invite_event.sender)
 
             await self.opsdroid.parse(
                 events.UserInvite(
                     target=roomid,
-                    user_id=invite_event["sender"],
+                    user_id=invite_event.sender,
                     user=sender,
                     connector=self,
                     raw_event=invite_event,
                 )
             )
 
-        for roomid, room in response["rooms"]["join"].items():
-            if "timeline" in room:
-                for event in room["timeline"]["events"]:
-                    if event["sender"] != self.mxid:
-                        return await self._event_creator.create_event(event, roomid)
+        for roomid, roomInfo in response.rooms.join.items():
+            if roomInfo.timeline:
+                for event in roomInfo.timeline.events:
+                    if event.sender != self.mxid:
+                        # if isinstance(event, nio.MegolmEvent):
+                        # event is encrypted
+                        # event = await self.connection.decrypt_event(event)
+
+                        return await self._event_creator.create_event(
+                            event.source, roomid
+                        )
```

Comment:
The event creator really needs the content dict, as it's supposed to set it as `raw_event` on the generated `opsdroid.Event`. Maybe it would be better to pass that through?

Reply:
You mean opsdroid expects `raw_event` to have a `content` dict? 
In that case we could do its as 
```
event.source["content"] = event.content
```
Or does content need more stuff in it?

Reply:
I mean `raw_event` should be the content dict. What is `event.source` and `event.content`? I am not really familiar with these nio classes.

Reply:
I think source is the info about the response and content is the actual body of the event. In other nio event classes content is part of source but not in this case, not sure why. For all the other events we've been passing source as raw_event since that's where the stuff like sender and event type is